gazebo_ros2_control/src/gazebo_system.cpp

```diff
@@ -551,10 +534,11 @@ GazeboSystem::perform_command_mode_switch(
     }
   }
 
-  // mimic joint has the same control mode as mimicked joint
-  for (const auto & mimic_joint : this->dataPtr->mimic_joints_) {
-    this->dataPtr->joint_control_methods_[mimic_joint.joint_index] =
-      this->dataPtr->joint_control_methods_[mimic_joint.mimicked_joint_index];
+  // mimic joint has always position control mode
+  for (const auto & mimic_joint : this->info_.mimic_joints) {
+    this->dataPtr->joint_control_methods_[mimic_joint.joint_index] &=
+      static_cast<ControlMethod_>(VELOCITY & EFFORT);
+    this->dataPtr->joint_control_methods_[mimic_joint.joint_index] |= POSITION;
```

Comment:
@christophfroehlich can you explain me what you are doing here?

You are making sure that we only have the control method in position but not in velocity and effort?

Reply:
Exactly, this is the code used elsewhere. I didn't manage to simply set it to `POSITION` first, because this is of type SafeEnum
`typedef SafeEnum<enum ControlMethod_> ControlMethod;`
but a `static_cast` seems to work now, see the last commit.

Reply:
Yes. Now it is very clear. Thanks